### PR TITLE
migrate to @noble/curves

### PR DESCRIPTION
**Pull request by lourkeur**
> Switch to @noble/curves for security updates

Right now signature verification fails. I think it's the DST but I cant figure out the right one.

```
  ● verifyBeacon › signatures on G1 › should verify a signature on G1

    expect(received).resolves.toEqual(expected) // deep equality

    Expected: true
    Received: false

      170 |
      171 |         it('should verify a signature on G1', async () => {
    > 172 |             await expect(verifyBeacon(chainInfo, validBeacon)).resolves.toEqual(true)
          |                                                                         ^
      173 |         })
      174 |
      175 |         it('should not verify a signature on G1 for the wrong round', async () => {

      at Object.toEqual (node_modules/expect/build/index.js:194:22)
      at Object.<anonymous> (test/beacon-verification.test.ts:172:73)
```

### CODE REVIEW

File: lib/beacon-verification.ts [hunk 1]
```diff
@@ -66,13 +71,13 @@ export async function verifySigOnG1(
...
+    const Hm = normP1Hash(message, domainSeparationTag);
+    const G = bls.G2.ProjectivePoint.BASE;
     const S = normP1(signature);
-    const ePHm = pairing(Hm, P.negate(), false);
-    const eGS = pairing(S, G, false);
-    const exp = eGS.multiply(ePHm).finalExponentiate();
-    return exp.equals(Fp12.ONE);
+    const ePHm = bls.pairing(Hm, P.negate(), false);
+    const eGS = bls.pairing(S, G, false);
+    const exp = bls.fields.Fp12.mul(eGS, ePHm);
```
Comment (CluEleSsUK) [inline lib/beacon-verification.ts:79]: Is the final exponentation no longer necessary with updated noble curves?
From what I can see the DST looks correct

Comment (lourkeur) [reply lib/beacon-verification.ts:79]: It could be that. I am not sure how to compare GT elements though...

File: lib/beacon-verification.ts [hunk 2]
```diff
@@ -66,13 +71,14 @@ export async function verifySigOnG1(
...
+    const G = bls.G2.ProjectivePoint.BASE;
     const S = normP1(signature);
-    const ePHm = pairing(Hm, P.negate(), false);
-    const eGS = pairing(S, G, false);
-    const exp = eGS.multiply(ePHm).finalExponentiate();
-    return exp.equals(Fp12.ONE);
+    const ePHm = bls.pairing(Hm, P.negate(), false);
+    const eGS = bls.pairing(S, G, false);
+    const Fp12 = bls.fields.Fp12 as any;// as {finalExponentiate(num: any): any;};
+    const exp = Fp12.finalExponentiate(bls.fields.Fp12.mul(eGS, ePHm));
```
Comment (lourkeur) [inline lib/beacon-verification.ts:80]: `finalExponentiate` is necessary. Tests pass, but we need to figure out a type-safe way to access it.

Comment (CluEleSsUK) [reply lib/beacon-verification.ts:80]: I've raised an issue on noble-curves: https://github.com/paulmillr/noble-curves/issues/65

If the build passes, I think we can merge this for now (after you remove the commented out bit) and I'll raise an issue to fix it if Paul gives us another route to do so or exposes it for us.

**Comment by AnomalRoil** [comment]
Can't you just set   to true the pairing argument that does final expo or doesn't do it?
`const ePHm = bls.pairing(Hm, P.negate(), true);`
we'll loose some performance but at least it should work, no?

**Comment by lourkeur** [comment]
> Can't you just set to true the pairing argument that does final expo or doesn't do it? `const ePHm = bls.pairing(Hm, P.negate(), true);` we'll lose some performance but at least it should work, no?

Yes it does. Performance-wise you're right on paper at least.

**Comment by lourkeur** [comment]
I'm on the `package-lock` issue

**Review by CluEleSsUK — APPROVED** [review]
Thank you so much for your contribution!!

**Pull request merged**